src/main/java/com/ericsson/ei/subscription/RunSubscription.java

```diff
@@ -113,9 +113,19 @@ public boolean runSubscriptionOnObject(String aggregatedObject, Iterator<JsonNod
             if (count_conditions != 0 && count_condition_fulfillment == count_conditions) {
                 conditionFulfilled = true;
                 if (subscriptionJson.get("repeat").toString() == "false" && id != null) {
-                    LOGGER.debug("Adding matched AggrObj id to SubscriptionRepeatFlagHandlerDb.");
-                    subscriptionRepeatDbHandler.addMatchedAggrObjToSubscriptionId(subscriptionName,
-                            requirementIndex, id);
+                    // the keyword 'synchronized' ensures that this part of the code run
+                    // synchronously. Thus avoids race condition.
+                    synchronized (this) {
+                        if (!subscriptionRepeatDbHandler.checkIfAggrObjIdExistInSubscriptionAggrIdsMatchedList(
+                                subscriptionName, requirementIndex, id)) {
+                            LOGGER.debug("Adding matched AggrObj id to SubscriptionRepeatFlagHandlerDb.");
+                            subscriptionRepeatDbHandler.addMatchedAggrObjToSubscriptionId(subscriptionName,
+                                    requirementIndex, id);
+                        } else {
+                            conditionFulfilled = false;
+                            break;
```

Comment:
This break will break all condition checks.
A subscription may have many requirements, and each contidion may contain several conditions. if the first requirement has been triggered it will never check the later ones.